File: src/main/java/com/clinic/domain/entities/CountryISO.java
package com.clinic.domain.entities;

/**
 * Supported countries for country-specific booking flows (PE / CL),
 * matching the AWS project's country-based processing.
 */
public enum CountryISO {
    PE,
    CL;

    public static boolean isSupported(String value) {
        if (value == null || value.isBlank()) {
            return false;
        }
        for (CountryISO country : values()) {
            if (country.name().equals(value)) {
                return true;
            }
        }
        return false;
    }

    public static String supportedValues() {
        StringBuilder values = new StringBuilder();
        for (CountryISO country : values()) {
            if (!values.isEmpty()) {
                values.append(",");
            }
            values.append(country.name());
        }
        return values.toString();
    }
}

File: clinic/domain/entities/country_iso.py
"""Supported countries for country-specific booking flows (PE / CL)."""

from enum import Enum


class CountryISO(str, Enum):
    PE = "PE"
    CL = "CL"

    @staticmethod
    def is_supported(value: str | None) -> bool:
        if value is None or value.strip() == "":
            return False
        return value in (c.value for c in CountryISO)

    @staticmethod
    def supported_values() -> str:
        return ",".join(c.value for c in CountryISO)
